thorlcr/master/thgraphmanager.cpp

```diff
@@ -331,10 +331,13 @@ void CJobManager::fatal(IException *e)
 
 void CJobManager::updateWorkUnitLog(IWorkUnit &workunit)
 {
-    StringBuffer log, logUrl;
+    StringBuffer log, logUrl, slaveLogPattern;
     logHandler->getLogName(log);
     createUNCFilename(log, logUrl, false);
-    workunit.addProcess("Thor", globals->queryProp("@name"), 0, logUrl.str());
+    slaveLogPattern.set(THORSLAVELOGSEARCHSTR).append(SLAVEIDSTR);
+    slaveLogPattern.append(strstr(log, THORMASTERLOGSEARCHSTR) + strlen(THORMASTERLOGSEARCHSTR) - 1); //Keep the '.' at the end of the MASTERLOGSEARCHSTR.
```

Comment:
that strstr should always match, but I think safer if you do on separate line and add a dbgassertex that it returned something.

Reply:
>at the end of the MASTERLOGSEARCHSTR

at the end of the THORLOGSEARCHSTR ?

Reply:
is this line effecttively just keeping the file extension? i.e. ".log" ?

Reply:
Will change to:
const char *ptr = strstr(log, THORMASTERLOGSEARCHSTR);
dbgassertex(ptr);
slaveLogPattern.append(ptr + strlen(THORMASTERLOGSEARCHSTR) - 1); //Keep the '.' at the end of the THORMASTERLOGSEARCHSTR.

The '.log' is in the pattern string. Ex: pattern="thorslave.SLAVEID.2020_07_23.log"

Reply:
ok